packages/shared/src/multisig.proxy.ts

```diff
@@ -0,0 +1,726 @@
+
+  // Check that Alice's multisig creation deposit was reserved.
+  let aliceReservedFunds = await getReservedFunds(client, alice.address)
+  expect(aliceReservedFunds, 'Alice should have reserved funds for multisig deposit').toBe(
+    await getMultisigCosts(client, threshold),
+  )
+
+  // Check the multisig creation event and extract multisig account address.
+  const [primaryMultisigAddress, primaryMultisigExtrinsicIndex] = await getAndVerifyMultisigEventData(
+    client,
+    alice.address,
+    addressEncoding,
+  )
+
+  // Funds the multisig account.
+  const extraFunds = 1e10
+  await setupBalances(client, [{ address: primaryMultisigAddress, amount: transferAmount + extraFunds }])
+
+  // Approve the multisig call. This is the final approval, so `multisig.asMulti` is used.
+  const primaryMultiLastTx = client.api.tx.multisig.asMulti(
+    threshold,
+    [alice.address, bob.address].sort(),
+    {
+      height: currBlockNumber + 1,
+      index: primaryMultisigExtrinsicIndex,
+    },
+    transferCall.method.toHex(),
+    primarymaxWeight,
+  )
+
+  const proxyTx = client.api.tx.proxy.proxy(pureProxyAddress, null, primaryMultiLastTx)
+  const secondaryMaxWeight = { refTime: 10000000000, proofSize: 10000000 } // Conservative weight limit
+
+  const secondaryMultiFirstTx = client.api.tx.multisig.asMulti(
+    threshold,
+    [dave.address, eve.address].sort(),
+    null, // No timepoint for first approval
+    proxyTx.method.toHex(),
+    secondaryMaxWeight,
+  )
+
+  await sendTransaction(secondaryMultiFirstTx.signAsync(charlie))
+  currBlockNumber = (await client.api.rpc.chain.getHeader()).number.toNumber()
+
+  await client.dev.newBlock()
+
+  // Check the multisig creation event (and extract multisig account address)
+  const [secondaryMultisigAddress, secondaryMultisigExtrinsicIndex] = await getAndVerifyMultisigEventData(
+    client,
+    charlie.address,
+    addressEncoding,
+  )
+
+  await setupBalances(client, [{ address: pureProxyAddress, amount: 100e10 }])
+
+  // Charlie makes the secondary multisig the co-owner of the pure proxy.
+  const addProxyTx = client.api.tx.proxy.addProxy(secondaryMultisigAddress, 'Any', 0)
+  let pureProxyTx = client.api.tx.proxy.proxy(pureProxyAddress, null, addProxyTx)
+
+  await sendTransaction(pureProxyTx.signAsync(charlie))
+  await client.dev.newBlock()
+
+  // Charlie removes himself as the co-owner of the pure proxy.
+  const removeProxyTx = client.api.tx.proxy.removeProxy(charlie.address, 'Any', 0)
+  pureProxyTx = client.api.tx.proxy.proxy(pureProxyAddress, null, removeProxyTx)
+
+  await sendTransaction(pureProxyTx.signAsync(charlie))
+  await client.dev.newBlock()
+
+  // Eve approves the secondary multisig call, which makes the pure proxy sign the first multisig call, thus funding Dave
+  const secondaryMultiLastTx = client.api.tx.multisig.asMulti(
+    threshold,
+    [charlie.address, dave.address].sort(),
+    {
+      height: currBlockNumber + 1,
```

Comment:
A note for the future: this works for multisigs.
However, for most other pallets (proxy, scheduler, vesting), the block provider that an asset hub runtime's pallet has access to is nonlocal i.e. the relay's.

This would promptly fail in those cases.

Just FYI.